src/test/java/software/amazon/awssdk/crt/test/Http2StreamManagerTest.java

```diff
@@ -167,7 +161,8 @@ public void testParallelRequestsWithLeakCheck(int numThreads, int numRequests) t
         // don't bother.
         // If we have a leak, we should have it on all platforms, and we'll catch it
         // elsewhere.
-        if (CRT.getOSIdentifier() != "android") {
+        // Disable the leak check for GraalVM as it's not using JVM, the assumption is not applicable.
+        if (CRT.getOSIdentifier() != "android" && System.getProperty("org.graalvm.nativeimage.imagecode") == null) {
```

Comment:
nit: disable comment is slightly confusing in this case as it is not enabled in the first place.